### PR TITLE
optimize: avoid concat string object, and kept source route id and se…

### PR DESCRIPTION
**Pull request by membphis**
fix https://github.com/iresty/apisix/issues/675

### CODE REVIEW

File: conf/config.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ nginx_config:                     # config for render the template to genarate n
 etcd:
   host: "http://127.0.0.1:2379"   # etcd address
   prefix: "/apisix"               # apisix configurations prefix
-  timeout: 1                      # 1 seconds
+  timeout: 3                      # 3 seconds
```
Comment (moonming) [inline conf/config.yaml]: Is this a related modification?

File: lua/apisix/plugin.lua [hunk 2]
```diff
@@ -281,53 +281,43 @@ end
 
 
 local function merge_service_route(service_conf, route_conf)
-    local new_service_conf
+    local new_conf = core.table.deepcopy(service_conf)
```
Comment (moonming) [inline lua/apisix/plugin.lua]: no test cases.

File: conf/config.yaml (hunk 1 above)
Comment (membphis) [reply conf/config.yaml]: the travis is slow now, we have to use a longer timeout.Otherwise the test case will randomly fail.

File: lua/apisix/plugin.lua (hunk 2 above)
Comment (membphis) [reply lua/apisix/plugin.lua]: I think the old test cases already cover this change.

**Comment by membphis** [comment]
@moonming please review again

File: conf/config.yaml (hunk 1 above)
Comment (moonming) [reply conf/config.yaml]: I think you should add issue for it, this method will affect all user, not only test cases.
